torch/distributed/pipeline/sync/pipe.py

```diff
@@ -117,21 +117,39 @@ def _retrieve_device(module: nn.Module) -> torch.device:
 
     return device if device is not None else torch.device("cpu")
 
+def _assemble_partition(modules: List[nn.Module]):
+    modules_list: List[nn.Module] = []
+    for module in modules:
+        if isinstance(module, nn.Sequential):
+            modules_list.extend(module.children())
```

Comment:
Why do we explicitly unwrap an nn.Sequential here? We can just append the nn.Sequential as-is right?

Reply:
That's needed for skip connections IIRC

Reply:
This is apparently needed for skip connections

Reply:
Could you elaborate a bit here? What exactly fails if we just add the nn.Sequential as-is?

Reply:
So if we do not flatten each partition (keep the aprtition as a sequence of sequences), then inspect_skip_layout will not collect skip connections correctly. That functions expects each module in the sequence is a Skippable module if it involves skip connections.